syne_tune/experiments/plotting.py

```diff
@@ -0,0 +1,664 @@
+            subplots_kwargs = subplots.kwargs
+            ncols = subplots_kwargs["ncols"]
+            nrows = subplots.kwargs["nrows"]
+            subplot_titles = subplots.titles
+            legend_no = [] if subplots.legend_no is None else subplots.legend_no
+            if not isinstance(legend_no, list):
+                legend_no = [legend_no]
+            title_each_figure = subplots.title_each_figure
+        else:
+            subplot_xlims = None
+            nrows = ncols = 1
+            subplots_kwargs = dict(nrows=nrows, ncols=ncols)
+            subplot_titles = None
+            legend_no = [0]
+            title_each_figure = False
+        if subplot_titles is None:
+            subplot_titles = [plot_params.title] + ["" * (ncols - 1)]
+        ylim = plot_params.ylim
+        xlim = plot_params.xlim  # Can be overwritten by ``subplot_xlims``
+        xlabel = plot_params.xlabel
+        ylabel = plot_params.ylabel
+        tick_params = plot_params.tick_params
+
+        plt.figure(dpi=plot_params.dpi)
+        figsize = (5 * ncols, 4 * nrows)
+        fig, axs = plt.subplots(**subplots_kwargs, squeeze=False, figsize=figsize)
+        for subplot_no, stats_subplot in enumerate(stats):
+            row = subplot_no % nrows
+            col = subplot_no // nrows
+            ax = axs[row, col]
+            # Plot curves in the order of ``setups``. Not all setups may feature in
+            # each of the subplots
+            for i, (curves, setup_name) in enumerate(zip(stats_subplot, setup_names)):
+                if curves is not None:
+                    color = f"C{i}"
+                    x = curves["time"]
+                    ax.plot(x, curves["aggregate"], color=color, label=setup_name)
+                    ax.plot(x, curves["lower"], color=color, alpha=0.4, linestyle="--")
+                    ax.plot(x, curves["upper"], color=color, alpha=0.4, linestyle="--")
+            if subplot_xlims is not None:
+                xlim = (0, subplot_xlims[subplot_no])
+            if xlim is not None:
+                ax.set_xlim(*xlim)
+            if ylim is not None:
+                ax.set_ylim(*ylim)
+            if xlabel is not None and row == nrows - 1:
+                ax.set_xlabel(xlabel)
+            if ylabel is not None and col == 0:
+                ax.set_ylabel(ylabel)
+            if tick_params is not None:
+                ax.tick_params(**tick_params)
+            if subplot_titles is not None:
+                if title_each_figure:
+                    ax.set_title(subplot_titles[subplot_no])
+                elif row == 0:
+                    ax.set_title(subplot_titles[col])
+            if plot_params.grid:
+                ax.grid(True)
+            if subplot_no in legend_no:
+                ax.legend()
+        plt.show()
+        return fig, axs
+
+    def plot(
+        self,
+        benchmark_name: Optional[str] = None,
+        plot_params: Optional[PlotParameters] = None,
```

Comment:
We do we have to pass plot_params here when we already pass it in the __init__ function?

Reply:
You do not, but you can. You can specify benchmark-dependent params here, in particular `ylim` and `metric`.